cmd/tegola/cmd/root.go

```diff
@@ -89,23 +60,31 @@ func rootCmdValidatePersistent(cmd *cobra.Command, _ []string) (err error) {
 		build.Commands = append(build.Commands, cmdName)
 		return nil
 	default:
-		return initConfig(configFile, requireCache, logLevel, logger)
+		return initConfig(configFile, requireCache, logLevel)
 	}
 }
 
-func initConfig(configFile string, cacheRequired bool, logLevel string, logger string) (err error) {
-	err = validateSupportedLoggers(logger)
-	if err != nil {
-		return err
-	}
-	log.SetLogger(logger)
-
-	// set log level before the first log is called
-	level, err := getLogLevelFromString(logLevel)
-	if err != nil {
-		return err
-	}
-	log.SetLogLevel(level)
+func initConfig(configFile string, cacheRequired bool, logLevel string) (err error) {
+	// Parse the provided log level; default to INFO if parsing fails.
+	lvl := log.ParseLogLevel(logLevel)
+
+	// Create a base text handler that outputs to stderr.
+	// The AddSource option includes file and line info in each log record.
+	baseHandler := slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
+		Level: lvl,
+		// TODO: enable once we switch to slog.Default
+		// instead of internal/log methods
+		AddSource: false,
+	})
```

Comment:
Consider moving this instantiation to the `internal/log` package. You're using it here and inside your tests so it might be better to centralize the code.

Reply:
Into an `init()`?